### PR TITLE
feat(tests): add baseline utility for integration testing from frontend ui

### PR DESCRIPTION
**Pull request by ntindle**
We want a way to automatically create and login to an account across the tests in a specific run

### Changes 🏗️
- Updates the creation and login pages/fixtures to store the user details in .auth/ folders
- Adds SIMPLE profile tests
- Finds bug in profile

#### For code changes:
- [x] I have clearly listed my changes in the PR description
- [x] I have made a test plan -> Writing tests!

### CODE REVIEW

File: autogpt_platform/frontend/src/tests/profile.spec.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    // await test
+    //   .expect(page.getByTestId("profile-section-personal"))
+    //   .toBeVisible();
+    // await test
+    //   .expect(page.getByTestId("profile-section-settings"))
+    //   .toBeVisible();
+    // await test
+    //   .expect(page.getByTestId("profile-section-security"))
+    //   .toBeVisible();
+  });
```
Comment (ntindle) [inline autogpt_platform/frontend/src/tests/profile.spec.ts:56]: To come later, with the rest of the real profile tests

File: autogpt_platform/frontend/src/tests/pages/navbar.page.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    await this.page.getByTestId("monitor-link").click();
+  }
+
+  async clickBuildLink() {
+    await this.page.getByTestId("build-link").click();
+  }
+
+  async clickMarketplaceLink() {
+    await this.page.getByTestId("marketplace-link").click();
+  }
```
Comment (ntindle) [inline autogpt_platform/frontend/src/tests/pages/navbar.page.ts:23]: These links aren't actually implemented yet. Will setup on shadcn to do these if I have time. If not, will manually update as needed.

File: autogpt_platform/frontend/src/tests/fixtures/index.ts [hunk 3]
```diff
@@ -1,18 +1,109 @@
...
+          throw signUpError;
+        }
+
+        auth.id = user?.id;
+        fs.writeFileSync(fileName, JSON.stringify(auth));
+      }
+
+      await use(auth);
+
+      // Cleanup code is commented out to preserve test users during development
```
Comment (aarushik93) [inline autogpt_platform/frontend/src/tests/fixtures/index.ts:83]: should it not run for CI purposes?

File: autogpt_platform/frontend/src/tests/profile.spec.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    await loginPage.login(testUser.email, testUser.password);
+    await test.expect(page).toHaveURL("/");
+  });
+
+  test("user can view their profile information", async ({
+    page,
+    testUser,
+  }) => {
+    await profilePage.navbar.clickProfileLink();
+    // sleep for 10 seconds to allow page to load due to bug in our system
```
Comment (aarushik93) [inline autogpt_platform/frontend/src/tests/profile.spec.ts:23]: what bug....?

File: autogpt_platform/frontend/src/tests/fixtures/index.ts (hunk 3 above)
Comment (ntindle) [reply autogpt_platform/frontend/src/tests/fixtures/index.ts:83]: ci makes a new db every time

File: autogpt_platform/frontend/src/tests/profile.spec.ts (hunk 4 above)
Comment (ntindle) [reply autogpt_platform/frontend/src/tests/profile.spec.ts:23]: the one where the profile page never loads after you login for the first time

**Comment by ntindle** [comment]
> > Finds bug in profile
> 
> what does this mean in changes?

That I found a bug in the profile page

File: autogpt_platform/frontend/src/tests/profile.spec.ts (hunk 4 above)
Comment (ntindle) [reply autogpt_platform/frontend/src/tests/profile.spec.ts:23]: #8788

**Review by aarushik93 — APPROVED** [review]

**Pull request merged**